### PR TITLE
Prepare API for parser validator improvements

### PR DESCRIPTION
**Pull request by BhaveshParmar00**

### CODE REVIEW

File: test/api/controllers/RequestHandlerSpec.scala [hunk 1]
```diff
@@ -121,72 +115,14 @@ class RequestHandlerSpec
 
     }
 
-    "given a request with a RequestCannotBeFulfilled gov-test-scenario header" when {
```
Comment (FergusHMRC) [inline test/api/controllers/RequestHandlerSpec.scala:124]: Why is this removed? missing context for it

Comment (BhaveshParmar00) [reply test/api/controllers/RequestHandlerSpec.scala:124]: Updated spec to include this

**Review by FergusHMRC — APPROVED** [review]
